src/game/shared/tf/tf_wearable.cpp

```diff
@@ -96,4 +96,47 @@ int C_TFWearable::InternalDrawModel( int flags )
 	return ret;
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: 
+//-----------------------------------------------------------------------------
+void C_TFWearable::UpdateModelToClass(void)
+{
+	if (m_bExtraWearable && m_Item.GetStaticData())
+	{
+		SetModel(m_Item.GetStaticData()->extra_wearable);
+	}
+	else
+	{
+		C_TFPlayer *pOwner = ToTFPlayer(GetOwnerEntity());
+
+		if (pOwner)
+		{
+			const char *pszModel;
+			if (pOwner->m_Shared.InCond(TF_COND_DISGUISED))
+			{
+				if (tf2v_Zombie_bug.GetBool())
```

Comment:
What exactly is the point of this

Reply:
oh it was a silly bug I was keeping, you can remove it if you would like but make sure to put the original line to fix it was `if (pOwner->m_Shared.InCond(TF_COND_DISGUISED) && pOwner->IsEnemyPlayer())
			{
				pszModel = m_Item.GetPlayerDisplayModel(pOwner->m_Shared.GetDisguiseClass());
			}
			else
			{
				pszModel = m_Item.GetPlayerDisplayModel(pOwner->GetPlayerClass()->GetClassIndex());
			}`

Reply:
how ever this does also fix model from showing when the spy is disguised to teammates when the convar is set to 0

Reply:
The 2 lines do exactly the same thing

Reply:
Well what's the result?

Reply:
nope not in testing, the model still appears when the spy is disguised when using `if (pOwner->m_Shared.InCond(TF_COND_DISGUISED) && pOwner->IsEnemyPlayer())` but just doesn't make it update to the disguise class model also to teammates. it becomes invisible with the other method for some reason not sure why.